tests/test_multipart.py

```diff
@@ -371,6 +372,9 @@ def test_multipart_encode_files_raises_exception_with_StringIO_content() -> None
         httpx.Request("POST", url, data={}, files=files)  # type: ignore
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="TemporaryFile on windows is binary mode"
+)
 def test_multipart_encode_files_raises_exception_with_text_mode_file() -> None:
     url = "https://www.example.com"
     with tempfile.TemporaryFile(mode="w") as upload:
```

Comment:
Perhaps we can resolve the test case like this?...

```suggestion
    # TemporaryFiles are always binary mode on windows.
    # For this test case where we need a text-mode file.
    # We make sure to create-and-write in binary mode first,
    # and then open the file in text mode.
    with tempfile.NamedTemporaryFile() as temp:
        temp.write(b'xxxxxxxxxx')
        with open(upload.name, mode="r") as upload:
            ...
```

Or else use a mock file-like object perhaps?

Reply:
this doesn't look right? what's `upload.name`

Reply:
do you mean `temp.name`? it's not possible to do this on windows.

Reply:
so I think another way is to use a `io.StringIO`